MH-13379, Simplify Mime Type Handling

lkiesow opened the pull request:
I glanced at the mime type handling code in Opencast and could not
resist but had to simplify it :D

lkiesow committed:
MH-13379, Simplify Mime Type Handling

I glanced at the mime type handling code in Opencast and could not
resist but had to simplify it :D

staubesv commented on modules/common/src/main/java/org/opencastproject/mediapackage/elementbuilder/AbstractElementBuilderPlugin.java:
```diff
@@ -143,15 +143,9 @@ else if (!file.isFile())
-      } else {
-        return false;
-      }
+    for (MimeType m : mimeTypes) {
```
I would rather recommend to replace all code of this method by 

```
  if (mimeTypes != null) {
    return checkMimeType(file, mimeTypes);
  } else {
    throw new IllegalStateException("Mime types have not been initialized");
  } 

```

;-)

staubesv replied:
@lkiesow Actually, it seems like the object instance variable "mimeTypes" is never initialised... does this method work at all?

staubesv commented on modules/common/src/main/java/org/opencastproject/mediapackage/elementbuilder/AbstractElementBuilderPlugin.java:
```diff
@@ -112,7 +112,7 @@ else if (!file.isFile())
       if (mimeTypes.length > 0) {
         for (MimeType m : mimeTypes) {
-          if (m.equals(MimeTypes.fromURL(file.toURI().toURL())))
+          if (m.equals(MimeTypes.fromString(file.getPath())))
```
Can you remove the null and length (lines 112 & 113) checks here, too?

staubesv commented on modules/common/src/main/java/org/opencastproject/util/MimeTypes.java:
```diff
@@ -122,58 +115,21 @@ private MimeTypes() {
-      isr = new InputStreamReader(is);
-      char[] chars = new char[1024];
-      int count = 0;
-      while ((count = isr.read(chars)) > 0) {
```
OMG!!!

staubesv reviewed (dismissed):
@lkiesow Can you please apply your simplification (remove null and length checks) in the method checkMimeType(File file) also in the method checkMimeType(File file, MimeType[] mimeTypes) and then make the first method just call the other one? I think that is worth the little effort...

lkiesow committed:
MH-13379, Delete More Unnecessary Code

This patch deletes more unused code related to mime type parsing and
processing.
